Before submitting a PR: XUnit test runner that can be used directly

Hi,

As [discussed on Twitter](https://twitter.com/breki74/status/1203632900069044224), I needed an XUnit runner that actually writes all of its output using XUnit's `ITestOutputHelper` and that can be used directly, not just through `Property` attributes, so I implemented one by (mostly) cannibalizing existing code from FsCheck: https://github.com/breki/demeton/blob/master/Demeton.Tests/BetterFsCheckRunner.fs

Here's a sample on how it is used: https://github.com/breki/demeton/blob/master/Demeton.Tests/PropertiesHelp.fs#L46

While I could add the whole code as a PR, I think it would be a better idea to discuss how best to integrate it - do I just merge the code with the existing XUnit runner or implement it separately?

Are there any docs describing the prerequisites needed to run the build?

I'm getting this warning:
```
warning NU1701: Package 'FSharp.Core 4.0.0.1' was restored using '.NETPortable,Version=v0.0,Profile=Profile259' 
instead of the project target framework '.NETStandard,Version=v1.0'. 
This package may not be fully compatible with your project.
```

Here's the result of `dotnet --info` on my machine: https://gist.github.com/breki/ee4e5d851995a0feb1c74fd33b6d46a3

Should I ignore these warnings as welI?
```
D:\src\breki-FsCheck\examples\FsCheck.Examples\FsCheck.Examples.fsproj.metaproj 
: warning MSB4078: The project file "examples\FsCheck.Examples\FsCheck.Examples.fsproj" is not supported by MSBuild and cannot be built.
```

Also, when I compile the code in VS (both in Debug and Release configurations), it fails with these kinds of errors:
```
D:\src\breki-FsCheck\src\FsCheck\Arbitrary.fs(510,40): error FS0001: 
The type 'Numerics.BigInteger' does not support a conversion to the type 'decimal'
1>D:\src\breki-FsCheck\src\FsCheck\Runner.fs(257,17): error FS0039: 
The value or constructor 'printf' is not defined. Maybe you want one of the following:   Printf   fprintf   sprintf   fprintfn   PrintfFormat
```

I tried building in Rider, maybe this error from the build log can shed more light:

```
"D:\src\breki-FsCheck\src\FsCheck\FsCheck.fsproj" (build target) (1) ->
(ResolvePackageAssets target) -> 
 D:\src\breki-FsCheck\src\FsCheck\FsCheck.fsproj : error NU1202: Package FSharp.Core 4.2.3 is not compatible with netstandard1.0 (.NETStandard,Version=v1.0). Package FSharp.Core 4.2.3 supports:
D:\src\breki-FsCheck\src\FsCheck\FsCheck.fsproj : error NU1202:   - net45 (.NETFramework,Version=v4.5)
D:\src\breki-FsCheck\src\FsCheck\FsCheck.fsproj : error NU1202:   - netstandard1.6 (.NETStandard,Version=v1.6)
```

--- a/src/FsCheck.Xunit/FsCheck.Xunit.fsproj
+++ b/src/FsCheck.Xunit/FsCheck.Xunit.fsproj
@@ -8,6 +8,7 @@
   </PropertyGroup>
   <ItemGroup>
     <Compile Include="AssemblyInfo.fs" />
+    <Compile Include="Runner.fs" />
     <Compile Include="PropertyAttribute.fs" />
     <Compile Include="CheckExtensions.fs" />
   </ItemGroup>

--- a/src/FsCheck.Xunit/PropertyAttribute.fs
+++ b/src/FsCheck.Xunit/PropertyAttribute.fs
@@ -3,10 +3,11 @@
 open System
 open System.Threading.Tasks
 
-open FsCheck
 open Xunit
 open Xunit.Sdk
 open Xunit.Abstractions
+
+open FsCheck
 
 type PropertyFailedException =
     inherit Exception

--- a/src/FsCheck.Xunit/Runner.fs
+++ b/src/FsCheck.Xunit/Runner.fs
@@ -1,0 +1,20 @@
+﻿namespace FsCheck.Xunit
+
+open FsCheck
+   
+/// A runner for FsCheck (i.e. that you can use as Config.Runner) which outputs
+/// to Xunit's given ITestOutputHelper.
+/// For example, { Config.QuickThrowOnFailure with Runner = TestOutputRunner(output) }
+type TestOutputRunner(output: Xunit.Abstractions.ITestOutputHelper) =
+    interface IRunner with
+        member _.OnStartFixture t =
+            output.WriteLine (Runner.onStartFixtureToString t)
+        member _.OnArguments (ntest, args, every) =
+            output.WriteLine (every ntest args)
+        member _.OnShrink(args, everyShrink) =
+            output.WriteLine (everyShrink args)
+        member _.OnFinished(name,testResult) =
+            let resultText = Runner.onFinishedToString name testResult
+            match testResult with
+            | TestResult.True _ -> resultText |> output.WriteLine
+            | _ -> failwithf "%s" resultText

